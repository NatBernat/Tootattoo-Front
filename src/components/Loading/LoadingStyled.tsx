import styled from "styled-components";

const LoadingStyled = styled.div`
  display: flex;
  justify-content: center;
  align-content: center;
  margin-top: 35vh;

  img {
    width: 150px;
    height: 0 auto;
    animation: spin 3s cubic-bezier(0.5, 0.25, 0.5, 0.75) infinite;
    cursor: wait;
  }
  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export default LoadingStyled;
