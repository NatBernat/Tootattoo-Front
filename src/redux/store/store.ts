import { configureStore } from "@reduxjs/toolkit";
import usersReducer from "../features/userSlice/userSlice";
import tattoosReducer from "../features/tattoosSlice/tattoosSlice";

const store = configureStore({
  reducer: {
    user: usersReducer,
    tattoos: tattoosReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
