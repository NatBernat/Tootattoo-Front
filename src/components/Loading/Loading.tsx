import LoadingStyled from "./LoadingStyled";

const Loading = (): JSX.Element => {
  return (
    <LoadingStyled>
      <img
        src={process.env.REACT_APP_FRONT_URL + "images/imageLogo.svg"}
        alt="Loader"
      />
    </LoadingStyled>
  );
};

export default Loading;
