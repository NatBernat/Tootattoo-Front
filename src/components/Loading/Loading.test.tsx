import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import store from "../../redux/store/store";
import Loading from "./Loading";

describe("Given a Loading Component", () => {
  describe("When it's rendered", () => {
    test("Then it should always match this snapshot", () => {
      const LoadingComponent = TestRenderer.create(
        <Provider store={store}>
          <Loading />
        </Provider>
      );

      expect(LoadingComponent).toMatchSnapshot();
    });
  });
});
