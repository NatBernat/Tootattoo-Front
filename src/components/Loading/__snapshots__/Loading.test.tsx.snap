// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Loading Component When it's rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ gwLzZc"
>
  <img
    alt="Loader"
    src="https://bernat-arcusa-front-final-project-202204-bcn.netlify.app/images/imageLogo.svg"
  />
</div>
`;
